torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -1123,3 +1123,32 @@ def _generate_logical_mesh(
                             logical_mesh_shape)  # type: ignore  # numpy 2.2
 
   return logical_mesh
+
+
+class MarkShardingFunction(torch.autograd.Function):
+  """
+  Autograd function to mark_sharding on intermediate tensors and the gradient
+  of the intermediate tensors during backward pass.
+
+  Usage:
+  new_tensor = MarkShardingFunction.apply(tensor, mesh, ('axis_1', 'axis_2'))
```

Comment:
WDYT adding a Python function wrapper in the `spmd` module? I think it could be convenient to have something like:

```
t = xs.mark_sharding_and_gradients(t, partition_spec)
```

next to the usual `mark_sharding` function, and we'll also be able to mention this function in the documentation of the `mark_sharding`, highlighting that the former doesn't shard gradients.

Reply:
Yup, definitely a better UX. Will send a PR today.